src/Kernel/client/ExecutionPathVisualizer/formatters/gateFormatter.ts

```diff
@@ -0,0 +1,308 @@
+    return formattedGates.flat().join('\n');
+};
+
+/**
+ * Takes in an operation's metadata and formats it into SVG.
+ * 
+ * @param metadata Metadata object representation of gate.
+ * 
+ * @returns SVG representation of gate.
+ */
+const _formatGate = (metadata: Metadata): string => {
+    const { type, x, controlsY, targetsY, label, argStr, width } = metadata;
+    switch (type) {
+        case GateType.Measure:
+            return _measure(x, controlsY[0], targetsY[0]);
+        case GateType.Unitary:
+            return _unitary(label, x, targetsY, width, argStr);
+        case GateType.Swap:
+            if (controlsY.length > 0) return _controlledGate(metadata);
+            else return _swap(x, targetsY);
+        case GateType.Cnot:
+        case GateType.ControlledUnitary:
+            return _controlledGate(metadata);
+        case GateType.ClassicalControlled:
+            return _classicalControlled(metadata);
+        default:
+            throw new Error(`ERROR: unknown gate (${label}) of type ${type}.`);
+    }
+};
+
+/**
+ * Creates a measurement gate at the x position, where qy and cy are
+ * the y coords of the qubit register and classical register, respectively.
+ * 
+ * @param x  x coord of measurement gate.
+ * @param qy y coord of qubit register.
+ * @param cy y coord of classical register.
+ * 
+ * @returns SVG representation of measurement gate.
+ */
+const _measure = (x: number, qy: number, cy: number): string => {
+    x -= minGateWidth / 2;
+    const width: number = minGateWidth, height = gateHeight;
+    // Draw measurement box
+    const mBox: string = box(x, qy - height / 2, width, height);
+    const mArc: string = arc(x + 5, qy + 2, width / 2 - 5, height / 2 - 8);
+    const meter: string = line(x + width / 2, qy + 8, x + width - 8, qy - height / 2 + 8);
+    const svg: string = group(mBox, mArc, meter);
+    return svg;
+};
+
+/**
+ * Creates the SVG for a unitary gate on an arbitrary number of qubits.
+ * 
+ * @param label            Gate label.
+ * @param x                x coord of gate.
+ * @param y                Array of y coords of registers acted upon by gate.
+ * @param width            Width of gate.
+ * @param argStr           Arguments passed in to gate.
+ * @param renderDashedLine If true, draw dashed lines between non-adjacent unitaries.
+ * 
+ * @returns SVG representation of unitary gate.
+ */
+const _unitary = (label: string, x: number, y: number[], width: number, argStr?: string, renderDashedLine: boolean = true): string => {
+    if (y.length === 0) return "";
+
+    // Sort y in ascending order
+    y.sort((y1, y2) => y1 - y2);
+
+    // Group adjacent registers
+    let prevY: number = y[0];
+    const regGroups: number[][] = y.reduce((acc: number[][], currY: number) => {
+        // Registers are defined to be adjacent if they differ by registerHeight in their y coord
+        // Raphael: Is there a better way of doing this? (i.e. split into y and heights in processOperations)
```

Comment:
Just a reminder that this note is here, in case you wanted to do something about it.

Reply:
Thanks for the heads-up! I think that this method works for now, but it's there because it might break in the future if we want to add variable heights for each register. I can edit this comment to be more of a note rather than a "note-to-self".